plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/callables/FunctionEmbedding.kt

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.embeddings.callables
+
+import org.jetbrains.kotlin.formver.viper.ast.Method
+
+interface FunctionEmbedding : CallableEmbedding {
```

Comment:
Are you planning to use `FunctionEmbedding` elsewhere, otherwise I don't see a reason why to make a distinction between `UserFunctionEmbedding` and potentially other functions.

Reply:
Yes: the plan is to move all the builtin functions we have into `FunctionEmbedding`, the way I wanted to do in https://github.com/jesyspa/kotlin/pull/93